models/src/test/scala/coop/rchain/models/EqualMSpec.scala

```diff
@@ -77,27 +82,34 @@ class EqualMSpec extends AnyFlatSpec with ScalaCheckPropertyChecks with Matchers
       forAllSimilarA[A]((x, y) => sameResultAsReference(x, y))
     }
 
+  // this @nowarn is to disable warning which are fatal according to the project configuration
+  // 'non-variable type argument Any in type pattern coop.rchain.models.AlwaysEqual[Any] is unchecked since it
+  // is eliminated by erasure [error] case (_: AlwaysEqual[Any], _: AlwaysEqual[Any]) => self.equals(other)'
+  @nowarn
   private def sameResultAsReference[A <: Any: EqualM: Pretty](x: A, y: A): Assertion = {
     // We are going to override the generated hashCode for our generated AST classes,
     // so in this test we rely on the underlying implementation from ScalaRuntime,
     // and hard-code the current definition for the handmade AST classes.
-
     def reference(self: Any, other: Any): Boolean = (self, other) match {
       case (left: ParSet, right: ParSet) =>
-        Equiv.by((x: ParSet) => (x.ps, x.remainder, x.connectiveUsed)).equiv(left, right)
+        val eqF = (x: ParSet) => (x.ps, x.remainder, x.connectiveUsed)
+        Equiv.universal.equiv(eqF(left), eqF(right))
       case (left: ParMap, right: ParMap) =>
-        Equiv.by((x: ParMap) => (x.ps, x.remainder, x.connectiveUsed)).equiv(left, right)
+        val eqF = (x: ParMap) => (x.ps, x.remainder, x.connectiveUsed)
+        Equiv.universal.equiv(eqF(left), eqF(right))
+      // this case is introduced after migration to scala 2.13 since AlwaysEqual falls under Product case for some reason
+      case (_: AlwaysEqual[Any], _: AlwaysEqual[Any]) => self.equals(other)
       case (left: Product, right: Product) =>
         left.getClass.isInstance(other) &&
           left.productIterator
             .zip(right.productIterator)
             .forall(tupled(reference))
-      case _ => self.equals(other)
+      case _ => self == other
```

Comment:
This one cannot be removed since there are test cases like `sameResultAsReference[Int]`, so input is a tuple if two integers and it falls under this case.